Content.Server/Construction/Completions/SetAnchor.cs

```diff
@@ -12,7 +13,17 @@ public sealed partial class SetAnchor : IGraphAction
         public void PerformAction(EntityUid uid, EntityUid? userUid, IEntityManager entityManager)
         {
             var transform = entityManager.GetComponent<TransformComponent>(uid);
-            transform.Anchored = Value;
+
+            if (Value == transform.Anchored)
+                return;
+
+            var mapManager = IoCManager.Resolve<IMapManager>();
+            var transformSystem = entityManager.System<SharedTransformSystem>();
+
+            if (Value && mapManager.TryFindGridAt(transformSystem.GetMapCoordinates(transform), out var gridUid, out var grid))
+                transformSystem.AnchorEntity((uid, transform), (gridUid, grid));
+            else
+                transformSystem.Unanchor(uid, transform);
         }
```

Comment:
It would be better for engine to do this instead of content having to manually do it every time, whether via TryAnchor or the likes.

Reply:
Moved to the engine.